packages/react/src/hooks/useExperiment.ts

```diff
@@ -29,7 +29,7 @@ export function useExperiment(key: string, defaultValue = false): boolean {
   const { gates } = useContext(GateContext) ?? {};
   const xp = gates?.store?.experiments ?? {};
 
-  const hashedKey = createHash(key);
+  const hashedKey = createHash(key, String(gates?.salt));
```

Comment:
The problem with `String()` casting this way, is that if `gates?.salt` is null/undefined, the result will be `"null" | "undefined"`

Reply:
`salt` would never be `null` or `undefined` given that we have a fallback on core, that's just bad type inference on the main interface